src/main/java/com/epam/test_generator/controllers/JiraController.java

```diff
@@ -0,0 +1,109 @@
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+/**
+ * Allow communication with Jira
+ */
+@RestController
+@RequestMapping(value = "/jira")
+public class JiraController {
+
+    @Autowired
+    private JiraService jiraService;
+
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_LEAD"})
+    @ApiImplicitParam(name = "Authorization", value = "add here your token", paramType = "header", dataType = "string", required = true)
+    @RequestMapping(value = "/projects", method = RequestMethod.GET, produces = "application/json")
+    public ResponseEntity<List<JiraProject>> getProjects() throws JiraException {
+
+        return new ResponseEntity<>(jiraService.getNonexistentJiraProjects(), HttpStatus.OK);
+    }
+
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_LEAD"})
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "jiraKey", value = "Key of project", required = true, dataType = "long", paramType = "path"),
+            @ApiImplicitParam(name = "Authorization", value = "add here your token", paramType = "header", dataType = "string", required = true)
+    })
+    @RequestMapping(value = "/project/{jiraKey}/stories", method = RequestMethod.GET, produces = "application/json")
+    public ResponseEntity<List<JiraStory>> getAllStories(
+            @PathVariable("jiraKey") String jiraProjectKey)
+            throws JiraException {
+
+        return new ResponseEntity<>(jiraService.getJiraStoriesFromJiraProjectByProjectId(jiraProjectKey), HttpStatus.OK);
+    }
+
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_LEAD"})
+    @ApiImplicitParam(name = "Authorization", value = "add here your token", paramType = "header", dataType = "string", required = true)
+    @RequestMapping(value = "/project", method = RequestMethod.POST, consumes = "application/json")
+    public ResponseEntity<String> createProjectWithAttFromJira(
+            @RequestBody(required = false) List<JiraStory> jiraStories,
+            Authentication auth) throws JiraException {
+
+        jiraService.createProjectWithAttachments(jiraStories, auth);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+//    @Secured({"ROLE_ADMIN", "ROLE_TEST_LEAD"})
+//    @ApiImplicitParams({
+//            @ApiImplicitParam(name = "jiraKey", value = "Key of project", required = true, dataType = "long", paramType = "path"),
+//            @ApiImplicitParam(name = "Authorization", value = "add here your token", paramType = "header", dataType = "string", required = true)})
+//    @RequestMapping(value = "project/{projectId}/stories/", method = RequestMethod.POST, produces = "application/json")
+//    public ResponseEntity<List<JiraStory>> getUnexistedStories(
+//            @PathVariable("jiraKey") String jiraProjectKey) throws JiraException {
+//
+//        return new ResponseEntity<>(jiraService.getJiraStoriesFromJiraProjectByProjectId(jiraProjectKey),
+//                HttpStatus.OK);
+//    }
```

Comment:
Actually you don't need to save commented code in a source file.
The old version will be safed by git.
However, if it is a new file you can,remove it after the first commit.

Reply:
+